src/main/resources/api/data-repository-openapi.yaml

```diff
@@ -1094,6 +1094,110 @@ paths:
             application/json:
               schema:
                 $ref: '#/components/schemas/ErrorModel'
+  /api/repository/v1/snapshots/{id}/linkDuosDataset/{duosId}:
+    put:
+      tags:
+        - snapshots
+        - repository
+      description: >
+        Link the DUOS dataset to the snapshot so that its authorized users are synced
+        as snapshot readers.
+        NOTE: This is an experimental feature and its response body may change.
```

Comment:
More context: right now we execute a synchronous flight and return the result, but it's possible we could prefer an asynchronous flight instead.  In that case, we'd return our usual job response which puts the onus on the caller to poll it to completion.